From 17b95df49eb83698f4bf0c81f96b84a5a2b259a6 Mon Sep 17 00:00:00 2001
From: Steve Schnepp <steve.schnepp@pwkf.org>
Date: Sun, 31 Jan 2016 10:11:12 +0000
Subject: [PATCH] asyncd: fix the 1h sleep

If waked up too early for all the plugins, it sleeps for $timeout,
which is 1h by default.

Backport to stable-2.0
---
 node/_bin/munin-asyncd.in | 8 +++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/node/_bin/munin-asyncd.in b/node/_bin/munin-asyncd.in
index 754d1a07..3d2a943a 100755
--- a/node/_bin/munin-asyncd.in
+++ b/node/_bin/munin-asyncd.in
@@ -148,8 +148,12 @@
 	PLUGIN: foreach my $plugin (@plugins) {
 		# See if this plugin should be updated
 		my $plugin_rate = $spoolwriter->get_metadata("plugin_rates/$plugin") || 300;
-		if ($when < ($last_updated{$plugin} || 0) + $plugin_rate) {
-			# not yet, next plugin
+		my $plugin_next = ($last_updated{$plugin} || 0) + $plugin_rate;
+		if ($when < $plugin_next) {
+			# not yet, wake me up later, and see next plugin
+			if ($plugin_next < $when_next) {
+				$when_next = $plugin_next;
+			}
 			next;
 		}
 
-- 
2.16.2

